crates/runtime/src/embeddings/connector.rs

```diff
@@ -453,16 +453,26 @@ impl DataConnector for EmbeddingConnector {
 }
 
 fn underlying_federated_table_for_indexed_table(
-    indexed_table: &Arc<IndexedTableProvider>,
+    src_table_provider: &Arc<dyn TableProvider>,
 ) -> Option<Arc<FederatedTable>> {
     #[cfg(feature = "s3_vectors")]
     {
-        let vector_scan = indexed_table
-            .underlying
+        if let Some(vector_scan) = src_table_provider
             .as_any()
-            .downcast_ref::<super::index::VectorScanTableProvider>()?;
+            .downcast_ref::<super::index::VectorScanTableProvider>()
```

Comment:
Instead of doing this, we should merge indexes together in the same provider when we're creating them.